server/db/schemas/addresses.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE addresses (
+    user_id INT NOT NULL,
```

Comment:
A single address row should only be tied to a single user id. Having multiple user ids tied to the same row is a database anti pattern because you are trying to de-duplicate data that is not duplicated.

What you are suggesting would couple the users together through an address which has no benefit and creates a lot of very complicated domain logic we would have to handle.

As it is now the domain model is satisfied where a user can have many addresses, there is some "duplication" in the sense that 2 people living at the same address would have almost identical rows but this is intentional to avoid the problems below and also make sense from the domain side because those users should not be coupled in any way.

Examples:
- If I create a new user with an address, we would have to first search to see if that address exists or not, if so just add the id to the join table, if not add it to the join table and to the address table
- If I want to delete my address, i have to make sure no other users are tied to that address then delete form the join table and the address table, else only delete from the join table
- You have the same problem for updating an address (assuming its not just a delete + add)